### PR TITLE
set default legend raster icon when wms layer

### PR DESCRIPTION
**Pull request by luipir**
## Description
Because of https://github.com/qgis/QGIS/pull/5299 that prevent segfault, the default icon in the legend list is empty. This PR restore default raster icon as WMS icon.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by luipir** [comment]
@m-kuhn follow up of https://github.com/qgis/QGIS/pull/5299

File: src/core/layertree/qgslayertreemodel.cpp [hunk 1]
```diff
@@ -227,6 +227,8 @@ QVariant QgsLayerTreeModel::data( const QModelIndex &index, int role ) const
         if ( testFlag( ShowRasterPreviewIcon ) )
         {
           QgsRasterLayer* rlayer = qobject_cast<QgsRasterLayer *>( layer );
+          if ( rlayer->providerType() == "wms" )
```
Comment (elpaso) [inline src/core/layertree/qgslayertreemodel.cpp]: QStringLiteral( "wms" )

Comment (m-kuhn) [reply src/core/layertree/qgslayertreemodel.cpp]: It's 2.18, but you can try QLatin1String ;)

Comment (luipir) [reply src/core/layertree/qgslayertreemodel.cpp]: +1 using QLatin1String would facilitate automatic porting.

**Comment by luipir** [comment]
sorry for my late... I was sick! any other comment @elpaso @m-kuhn ?

**Review by elpaso — APPROVED** [review]
LGTM

**Pull request merged**